File: MealSaver/src/main/java/com/marcu/mealsaver/Dto/RecipeDTO.java
package com.marcu.mealsaver.Dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RecipeDTO {
    private String title;
    private List<String> cleanedIngredients;
    private String instructions;
    private List<String> categories;
    private String imageName;
}

File: meal_saver_phone/meal_saver_phone/lib/models/recipe_dto.dart
class RecipeDTO {
  final String title;
  final List<String> cleanedIngredients;
  final String instructions;
  final List<String> categories;
  final String imageName;
  final List<String> sources;

  RecipeDTO({
    required this.title,
    required this.cleanedIngredients,
    required this.instructions,
    required this.categories,
    required this.imageName,
    required this.sources,
  });

  factory RecipeDTO.fromJson(Map<String, dynamic> json) {
    return RecipeDTO(
      title: json['title'],
      cleanedIngredients: List<String>.from(json['cleanedIngredients']),
      instructions: json['instructions'],
      categories: List<String>.from(json['categories']),
      imageName: json['imageName'],
      sources: List<String>.from(json['sources'] ?? []),
    );
  }
}
